src/datanode/src/heartbeat.rs

```diff
@@ -92,16 +99,31 @@ impl HeartbeatTask {
         let server_addr = self.server_addr.clone();
         let meta_client = self.meta_client.clone();
 
+        let catalog_manager_clone = self.catalog_manager.clone();
         let mut tx = Self::create_streams(&meta_client, running.clone()).await?;
         common_runtime::spawn_bg(async move {
             while running.load(Ordering::Acquire) {
+                let region_num = match region_number(catalog_manager_clone.clone()) {
```

Comment:
Also you can skip a clone by change `region_number`'s signature to `pub fn region_number(catalog_manager: &CatalogManagerRef) -> Result<u64>` and pass `&catalog_manager_clone`.